src/client/activation/activationManager.ts

```diff
@@ -56,10 +62,15 @@ export class ExtensionActivationManager implements IExtensionActivationManager {
         await this.initialize();
 
         // Activate all activation services together.
-        await Promise.all([
-            Promise.all(this.singleActivationServices.map((item) => item.activate())),
-            this.activateWorkspace(this.activeResourceService.getActiveResource()),
-        ]);
+
+        if (this.workspaceService.isVirtualWorkspace) {
+            await this.activateWorkspace(this.activeResourceService.getActiveResource());
+        } else {
+            await Promise.all([
+                Promise.all(this.singleActivationServices.map((item) => item.activate())),
```

Comment:
Nit, but I think you can write:

```suggestion
                ...this.singleActivationServices.map((item) => item.activate()),
```

To not nest promises. Probably doesn't matter.